### ISSUE TITLE
[Due for payment 2025-03-21] [$500] Expense - Green dot disappears from LHN when paying one out of many unpaid reports

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34767]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.27-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Precondition: There is a Collect workspace on Old Dot with admin and employee
pre-steps to have multiple reports for testing - 

- Go to the workspace settings > More features > Enable Workflows
- Go to the Workflows tab in the LHN
- Enable Delay submissions
- Keep the default workflow of Submit and Close

1. [Employee] Create a manual request in the workspace chat and submit it
2. [Employee] Create another manual request and submit it
3. {Admin] Go to workspace chat with employee
4. [Admin] Pay the second submitted report (submitted in Step 2)
5. Note that the LHN does not show green dot
6. [Admin] Click on the preview of first submitted report (Step 1) and return to the main chat

## Expected Result:
In Step 4, green dot should still appear for the workspace chat in LHN because there is still unpaid report

## Actual Result:
In Step 4, LHN for the workspace chat with unpaid report does not show green dot after paying one of the reports. It only shows green dot when revisiting main chat from the expense report

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/9fa5d1b7-103b-457a-8c67-b280b0ddeba8

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01fffa54d6d963be62</li>
        <li>Upwork Job ID: 1748072099859210240</li>
        <li>Last Price Increase: 2025-03-22</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @slafortune</details>

**Comment by lanitochka17** [comment]
[#wave6-collect-submitters](https://expensify.enterprise.slack.com/archives/C02MW39LT9N)

**Comment by allroundexperts** [comment]
I think Option 2 from @tienifr's [proposal](https://github.com/Expensify/App/issues/34767#issuecomment-1899797959) should work fine and be inline with the code that we are already using.

🎀 👀 🎀 C+ reviewed

**Comment by peterdbarkerUK** [comment]
[Confirming priority in slack](https://expensify.slack.com/archives/C02MW39LT9N/p1705936748767809)

**Comment by mountiny** [comment]
@roryabraham I believe to implement this properly, we would need to create App version of [this Auth method](https://github.com/Expensify/Auth/blob/27371b7cd306fb669658d71e159a5c7510fe30e2/auth/lib/ReportItem.cpp#L28-L71) and make sure we use this in backend when the report is reimbursed to update the GBR correctly.

**Comment by roryabraham** [comment]
> I believe to implement this properly, we would need to create App version of [this Auth method](https://github.com/Expensify/Auth/blob/27371b7cd306fb669658d71e159a5c7510fe30e2/auth/lib/ReportItem.cpp#L28-L71) and make sure we use this in backend when the report is reimbursed to update the GBR correctly.

we do have [ReportUtils::requiresAttentionFromCurrentUser](https://github.com/Expensify/App/blob/6e2f058b71d47a67397738054daee4604d753593/src/libs/ReportUtils.ts#L1769) which seems about the same – I wish these didn't have to be re-implemented and we could just have some shared logic in a C++ function that we could use in both places, but that's probably out-of-scope for this issue...

> Fix the back-end to return hasOutstandingChildRequest: true properly for the unpaid IOU report (currently it's returning hasOutstandingChildRequest: false even though those IOU reports are unpaid)

This sounds like it needs to be fixed, regardless. I think I prefer @tienifr's solution #1. I'm happy to assign him, but he'll be blocked on the back-end change.

Now I'm starting to think more seriously about creating some shared C++ libs for non-trivial business logic like this that's very input/output oriented and needs to be kept in sync across layers. But as that would be a new pattern, we need to discuss it further.

Sorry I don't have a more decisive conclusion here just yet – it's very late for me so I want to look at this one more time with fresher eyes.

**Comment by allroundexperts** [comment]
Hi @roryabraham!
Just following up on above. Did you get a chance to look at this again with a fresh pair of eyes?

**Comment by roryabraham** [comment]
Making this internal to implement @tienifr's proposed solution above, and adding this to wave7 as it's a bug for reimbursers.

**Comment by roryabraham** [comment]
We'll add `Help Wanted` back and hire @tienifr for the second half of solution number 1 of [this proposal](https://github.com/Expensify/App/issues/34767#issuecomment-1899797959) once the back-end change is made.

**Comment by slafortune** [comment]
Hey @roryabraham is there a link to the changes we're waiting on?

**Comment by slafortune** [comment]
I think while we wait, moving this to weekly works.

**Comment by roryabraham** [comment]
Ah, sorry for the delayed update. I don't have a back-end PR up yet

**Comment by roryabraham** [comment]
Leaving some breadcrumbs for myself to follow:

- There are 5 related API commands: https://github.com/Expensify/Web-Expensify/blob/886e1abe7f16eb020d95263322a96111e51b20e5/api.php#L3125-L3134
- `hasOutstandingChildRequest` is hard-coded to `false` [here](https://github.com/Expensify/Web-Expensify/blob/df07f0f74c0e56903cdfb9656f205d1b35fc4860/lib/ReportAPI.php#L5175)

**Comment by roryabraham** [comment]
No update right now, hopefully should be able to get this sorted this week though

**Comment by roryabraham** [comment]
no update, have been unable to prioritize anything other than NewDot deploy this week so far

**Comment by roryabraham** [comment]
Currently prioritizing getting the HybridApp build working again in https://github.com/Expensify/App/issues/36717

**Comment by slafortune** [comment]
Will be retested this week

**Comment by roryabraham** [comment]
I've got a number of things I'm planning to prioritize over this:

- reviewing 4 or 5 PRs on my plate
- last bit of https://github.com/Expensify/Expensify/issues/358172
- https://github.com/Expensify/Auth/pull/10072

then this will be next up. Hopefully that prioritization makes sense. If we need to, we can have someone else take over because I don't have a WIP PR up yet

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.

In Step 4, LHN for the workspace chat with unpaid report does not show green dot after paying one of the reports

https://github.com/Expensify/App/assets/43995119/e8da101b-87b9-455e-afac-374024396064

**Comment by roryabraham** [comment]
no update, currently prioritizing fires, deploy blockers, and (soon) CP-to-prod

**Comment by roryabraham** [comment]
No update, was focused on May 1st issue then [this important app reliability problem](https://github.com/Expensify/App/issues/41254)

**Comment by slafortune** [comment]
@roryabraham I'll be on vacation until June 4th - I'm going to leave this assigned to myself since it seems this is held up on you and no payment will be required before I am back.  If something changes, please remove me and readd the BZ label.

**Comment by slafortune** [comment]
Hey @roryabraham do you have a quick update on this?

**Comment by slafortune** [comment]
@roryabraham should this be left to weekly?

**Comment by trjExpensify** [comment]
I'm curious if this has been fixed along the way in GBR improvements?

**Comment by slafortune** [comment]
Ah!  This is outdated, request money is now submit expense.  Sorry I missed that, as it stands I can't recreate this and with submit expense I can recreate this.

**Comment by trjExpensify** [comment]
Sorry, I don't understand this: 

>Sorry I missed that, as it stands I can't recreate this and with submit expense I can recreate this.

**Comment by trjExpensify** [comment]
Cool, thanks. Seems like we still have to look into this then, @roryabraham. On the basis of [this](https://github.com/Expensify/App/issues/34767#issuecomment-2276975364), I'm moving it back to weekly as we focus on polish for #wave-collect.

**Comment by roryabraham** [comment]
It's not clear how to reproduce this now, since the two submitted manual requests are batched together and it's not clear how to pay the 2nd one without paying the 1st:

<img width="1141" alt="image" src="https://github.com/user-attachments/assets/b3c90e2d-4981-4161-acdd-09e7d318b312">

**Comment by roryabraham** [comment]
Same deal with P2P requests:

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/0a0b4adc-a7a1-43ab-a4ad-6980a23af8fa">

so this is indeed no longer reproducible

**Comment by roryabraham** [comment]
@tienifr feel free to provide updated reproduction steps if you'd like

**Comment by trjExpensify** [comment]
I suspect you had the workspace set to instant submit @roryabraham, so this needs a coupla' pre-steps to have multiple reports:

- Go to the workspace settings > More features > Enable `Workflows`
- Go to the Workflows tab in the LHN
- Enable `Delay submissions`

**Comment by slafortune** [comment]
Yes - I can still recreate this when setting delay submission.  I've updated the test steps to include that and leaving submit and close.

**Comment by roryabraham** [comment]
chatted with @trjExpensify [in slack](https://expensify.slack.com/archives/C036QM0SLJK/p1725297554913649?thread_ts=1725284076.327919&cid=C036QM0SLJK), it's unlikely I'll be able to prioritize this one this week so we may look for another volunteer

**Comment by slafortune** [comment]
Looking for a volunteer here - https://expensify.slack.com/archives/C049HHMV9SM/p1725553790788449

**Comment by roryabraham** [comment]
<details>
<summary><code>PayMoneyRequest</code> sample response</summary>

```
{
    "onyxData": [
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "participants": {
                    "4": {
                        "hidden": true,
                        "notificationPreference": "hidden"
                    }
                }
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "chatType": "",
                "description": "",
                "lastActorAccountID": 4,
                "lastMessageText": "marked this report as reimbursed",
                "lastVisibleActionCreated": "2024-09-16 22:15:39.923",
                "managerID": 4,
                "ownerAccountID": 4,
                "parentReportActionID": "7465655876950814176",
                "parentReportID": "616431942157016",
                "policyID": "5F78EE68DC8314F9",
                "reportID": "2881610082725380",
                "reportName": "Expense Report #2881610082725380",
                "state": "MANUALREIMBURSED",
                "stateNum": 2,
                "statusNum": 2,
                "type": "expense",
                "visibility": null
            }
        },
        {
            "key": "reportActions_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "1708872886801724526": {
                    "actionName": "MARKEDREIMBURSED",
                    "actorAccountID": 4,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "created": "2024-09-16 22:15:39.923",
                    "lastModified": "2024-09-16 22:15:39.923",
                    "message": [
                        {
                            "html": "",
                            "text": "",
                            "type": "COMMENT",
                            "whisperedTo": []
                        }
                    ],
                    "originalMessage": {
                        "lastModified": "2024-09-16 22:15:39.923",
                        "type": "Elsewhere"
                    },
                    "person": [
                        {
                            "style": "strong",
                            "text": "RoryTest Admin",
                            "type": "TEXT"
                        }
                    ],
                    "reportActionID": "1708872886801724526",
                    "shouldShow": true
                }
            }
        },
        {
            "key": "personalDetailsList",
            "onyxMethod": "merge",
            "value": {
                "4": {
                    "accountID": 4,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "displayName": "RoryTest Admin",
                    "firstName": "RoryTest",
                    "lastName": "Admin",
                    "login": "roryabrahamtest+admin@gmail.com",
                    "phoneNumber": "",
                    "pronouns": "",
                    "status": null,
                    "timezone": {
                        "automatic": false,
                        "selected": "America\/Los_Angeles"
                    },
                    "validated": true
                }
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "statusNum": 4
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "stateNum": 2
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "participants": {
                    "4": {
                        "hidden": true,
                        "notificationPreference": "hidden"
                    }
                }
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "chatType": "",
                "description": "",
                "lastActorAccountID": 4,
                "lastMessageText": "paid $20.00 elsewhere",
                "lastVisibleActionCreated": "2024-09-16 22:15:39.925",
                "managerID": 4,
                "ownerAccountID": 4,
                "parentReportActionID": "7465655876950814176",
                "parentReportID": "616431942157016",
                "policyID": "5F78EE68DC8314F9",
                "reportID": "2881610082725380",
                "reportName": "Expense Report #2881610082725380",
                "state": "MANUALREIMBURSED",
                "stateNum": 2,
                "statusNum": 4,
                "type": "expense",
                "visibility": null
            }
        },
        {
            "key": "reportActions_2881610082725380",
            "onyxMethod": "merge",
            "shouldShowPushNotification": true,
            "value": {
                "433572443584605772": {
                    "actionName": "IOU",
                    "actorAccountID": 4,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "created": "2024-09-16 22:15:39.925",
                    "lastModified": "2024-09-16 22:15:39.925",
                    "message": [
                        {
                            "html": "paid $20.00 elsewhere",
                            "text": "paid $20.00 elsewhere",
                            "type": "COMMENT",
                            "whisperedTo": []
                        }
                    ],
                    "originalMessage": {
                        "IOUReportID": 2881610082725380,
                        "amount": 2000,
                        "currency": "USD",
                        "entryID": null,
                        "lastModified": "2024-09-16 22:15:39.925",
                        "paymentType": "Elsewhere",
                        "type": "pay"
                    },
                    "person": [
                        {
                            "style": "strong",
                            "text": "RoryTest Admin",
                            "type": "TEXT"
                        }
                    ],
                    "reportActionID": "433572443584605772",
                    "shouldShow": true
                }
            }
        },
        {
            "key": "personalDetailsList",
            "onyxMethod": "merge",
            "value": {
                "4": {
                    "accountID": 4,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "displayName": "RoryTest Admin",
                    "firstName": "RoryTest",
                    "lastName": "Admin",
                    "login": "roryabrahamtest+admin@gmail.com",
                    "phoneNumber": "",
                    "pronouns": "",
                    "status": null,
                    "timezone": {
                        "automatic": false,
                        "selected": "America\/Los_Angeles"
                    },
                    "validated": true
                }
            }
        },
        {
            "key": "reportActions_616431942157016",
            "onyxMethod": "merge",
            "value": {
                "7465655876950814176": {
                    "actionName": "REPORTPREVIEW",
                    "actorAccountID": 4,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "childCommenterCount": 0,
                    "childLastActorAccountID": 4,
                    "childLastMoneyRequestComment": "",
                    "childLastReceiptTransactionIDs": "",
                    "childLastVisibleActionCreated": "",
                    "childMoneyRequestCount": 1,
                    "childOldestFourAccountIDs": "",
                    "childRecentReceiptTransactionIDs": [],
                    "childReportNotificationPreference": "hidden",
                    "childType": "expense",
                    "childVisibleActionCount": 0,
                    "created": "2024-09-16 22:15:17.788",
                    "lastModified": "2024-09-16 22:15:17.788",
                    "message": [
                        {
                            "html": "paid $20.00 elsewhere",
                            "text": "paid $20.00 elsewhere",
                            "type": "COMMENT",
                            "whisperedTo": []
                        }
                    ],
                    "originalMessage": {
                        "lastModified": "2024-09-16 22:15:17.788",
                        "linkedReportID": "2881610082725380"
                    },
                    "person": [
                        {
                            "style": "strong",
                            "text": "RoryTest Admin",
                            "type": "TEXT"
                        }
                    ],
                    "reportActionID": "7465655876950814176",
                    "shouldShow": true
                }
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "isCancelledIOU": false
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "lastReadTime": "2024-09-16 22:15:39.925"
            }
        },
        {
            "key": "report_616431942157016",
            "onyxMethod": "merge",
            "value": {
                "hasOutstandingChildRequest": false,
                "iouReportID": null
            }
        },
        {
            "key": "report_616431942157016",
            "onyxMethod": "merge",
            "value": {
                "lastActorAccountID": 4,
                "lastMessageText": "paid $20.00 elsewhere",
                "lastVisibleActionCreated": "2024-09-16 22:15:17.788",
                "maxSequenceNumber": null,
                "reportID": "616431942157016"
            }
        },
        {
            "key": "reportActions_616431942157016",
            "onyxMethod": "merge",
            "shouldNotify": false,
            "value": {
                "7465655876950814176": {
                    "actionName": "REPORTPREVIEW",
                    "actorAccountID": 4,
                    "automatic": false,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "childLastActorAccountID": 4,
                    "childManagerAccountID": 4,
                    "childMoneyRequestCount": 1,
                    "childOwnerAccountID": 4,
                    "childReportID": "2881610082725380",
                    "childReportName": "Expense Report #2881610082725380",
                    "childReportNotificationPreference": "hidden",
                    "childStateNum": 2,
                    "childStatusNum": 4,
                    "childType": "expense",
                    "created": "2024-09-16 22:15:17.788",
                    "isAttachment": false,
                    "lastModified": "2024-09-16 22:15:17.788",
                    "message": [
                        {
                            "deleted": "",
                            "html": "paid $20.00 elsewhere",
                            "isDeletedParentAction": false,
                            "isEdited": false,
                            "reactions": [],
                            "text": "paid $20.00 elsewhere",
                            "type": "COMMENT",
                            "whisperedTo": []
                        }
                    ],
                    "originalMessage": {
                        "lastModified": "2024-09-16 22:15:17.788",
                        "linkedReportID": "2881610082725380"
                    },
                    "person": [
                        {
                            "style": "strong",
                            "text": "RoryTest Admin",
                            "type": "TEXT"
                        }
                    ],
                    "reportActionID": "7465655876950814176",
                    "reportActionTimestamp": 1726524917788,
                    "sequenceNumber": null,
                    "shouldShow": true,
                    "timestamp": 1726524917,
                    "whisperedToAccountIDs": []
                }
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "avatarUrl": null,
                "chatReportID": "616431942157016",
                "chatType": "",
                "currency": "USD",
                "description": "",
                "errorFields": {
                    "export": null,
                    "notFound": null
                },
                "fieldList": null,
                "hasOutstandingChildRequest": false,
                "hasOutstandingChildTask": false,
                "hasParentAccess": true,
                "invoiceReceiver": null,
                "iouReportID": null,
                "isCancelledIOU": false,
                "isDeletedParentAction": null,
                "isOwnPolicyExpenseChat": false,
                "isPinned": false,
                "isWaitingOnBankAccount": false,
                "lastActionType": "IOU",
                "lastActorAccountID": "4",
                "lastMessageHtml": "paid $20.00 elsewhere",
                "lastMessageText": "paid $20.00 elsewhere",
                "lastReadSequenceNumber": 0,
                "lastVisibleActionCreated": "2024-09-16 22:15:39.925",
                "lastVisibleActionLastModified": "2024-09-16 22:15:39.925",
                "managerID": 4,
                "nonReimbursableTotal": 0,
                "notificationPreference": "hidden",
                "oldPolicyName": "",
                "ownerAccountID": 4,
                "parentReportActionID": "7465655876950814176",
                "parentReportID": "616431942157016",
                "participants": {
                    "4": {
                        "hidden": true,
                        "notificationPreference": "hidden"
                    }
                },
                "permissions": [],
                "policyAvatar": null,
                "policyID": "5F78EE68DC8314F9",
                "policyName": null,
                "private_isArchived": "",
                "reportID": "2881610082725380",
                "reportName": "Expense Report #2881610082725380",
                "stateNum": 2,
                "statusNum": 4,
                "total": -2000,
                "tripData": null,
                "type": "expense",
                "unheldTotal": -2000,
                "visibility": null,
                "welcomeMessage": "",
                "writeCapability": "all"
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "lastActorAccountID": 4,
                "lastMentionedTime": null,
                "lastMessageText": "You",
                "lastReadTime": "2024-09-16 22:15:39.925",
                "lastVisibleActionCreated": "2024-09-16 22:15:39.923",
                "maxSequenceNumber": 3,
                "reportID": "2881610082725380"
            }
        },
        {
            "key": "reportActions_2881610082725380",
            "onyxMethod": "merge",
            "shouldNotify": false,
            "value": {
                "1708872886801724526": {
                    "actionName": "MARKEDREIMBURSED",
                    "actorAccountID": 4,
                    "automatic": false,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "created": "2024-09-16 22:15:39.923",
                    "isAttachment": false,
                    "lastModified": "2024-09-16 22:15:39.923",
                    "message": [
                        {
                            "style": "strong",
                            "text": "You",
                            "type": "TEXT"
                        },
                        {
                            "style": "normal",
                            "text": " marked this report as manually reimbursed. This report will be reimbursed outside of Expensify",
                            "type": "TEXT"
                        }
                    ],
                    "originalMessage": {
                        "lastModified": "2024-09-16 22:15:39.923",
                        "type": "Elsewhere"
                    },
                    "person": [
                        {
                            "style": "strong",
                            "text": "RoryTest Admin",
                            "type": "TEXT"
                        }
                    ],
                    "reportActionID": "1708872886801724526",
                    "reportActionTimestamp": 1726524939923,
                    "sequenceNumber": 3,
                    "shouldShow": false,
                    "timestamp": 1726524939,
                    "whisperedToAccountIDs": []
                }
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "lastActorAccountID": 4,
                "lastMentionedTime": null,
                "lastMessageText": "paid $20.00 elsewhere",
                "lastReadTime": "2024-09-16 22:15:39.925",
                "lastVisibleActionCreated": "2024-09-16 22:15:39.925",
                "maxSequenceNumber": 4,
                "reportID": "2881610082725380"
            }
        },
        {
            "key": "reportActions_2881610082725380",
            "onyxMethod": "merge",
            "shouldNotify": false,
            "value": {
                "433572443584605772": {
                    "actionName": "IOU",
                    "actorAccountID": 4,
                    "automatic": false,
                    "avatar": "https:\/\/d2g02b6ed2w9fz.cloudfront.net\/7d759bc82ab83bbf508cb84009e17d654532c700_128.jpeg",
                    "created": "2024-09-16 22:15:39.925",
                    "isAttachment": false,
                    "lastModified": "2024-09-16 22:15:39.925",
                    "message": [
                        {
                            "deleted": "",
                            "html": "paid $20.00 elsewhere",
                            "isDeletedParentAction": false,
                            "isEdited": false,
                            "reactions": [],
                            "text": "paid $20.00 elsewhere",
                            "type": "COMMENT",
                            "whisperedTo": []
                        }
                    ],
                    "originalMessage": {
                        "IOUReportID": 2881610082725380,
                        "amount": 2000,
                        "currency": "USD",
                        "entryID": null,
                        "lastModified": "2024-09-16 22:15:39.925",
                        "paymentType": "Elsewhere",
                        "type": "pay"
                    },
                    "person": [
                        {
                            "style": "strong",
                            "text": "RoryTest Admin",
                            "type": "TEXT"
                        }
                    ],
                    "reportActionID": "433572443584605772",
                    "reportActionTimestamp": 1726524939925,
                    "sequenceNumber": 4,
                    "shouldShow": true,
                    "timestamp": 1726524939,
                    "whisperedToAccountIDs": []
                }
            }
        },
        {
            "key": "report_2881610082725380",
            "onyxMethod": "merge",
            "value": {
                "avatarUrl": null,
                "chatReportID": "616431942157016",
                "chatType": "",
                "currency": "USD",
                "description": "",
                "errorFields": {
                    "export": null,
                    "notFound": null
                },
                "fieldList": null,
                "hasOutstandingChildRequest": false,
                "hasOutstandingChildTask": false,
                "hasParentAccess": true,
                "invoiceReceiver": null,
                "iouReportID": null,
                "isCancelledIOU": false,
                "isDeletedParentAction": null,
                "isOwnPolicyExpenseChat": false,
                "isPinned": false,
                "isWaitingOnBankAccount": false,
                "lastActionType": "IOU",
                "lastActorAccountID": "4",
                "lastMentionedTime": null,
                "lastMessageHtml": "paid $20.00 elsewhere",
                "lastMessageText": "paid $20.00 elsewhere",
                "lastReadSequenceNumber": 0,
                "lastReadTime": "2024-09-16 22:15:39.925",
                "lastVisibleActionCreated": "2024-09-16 22:15:39.925",
                "lastVisibleActionLastModified": "2024-09-16 22:15:39.925",
                "managerID": 4,
                "nonReimbursableTotal": 0,
                "notificationPreference": "hidden",
                "oldPolicyName": "",
                "ownerAccountID": 4,
                "parentReportActionID": "7465655876950814176",
                "parentReportID": "616431942157016",
                "participants": {
                    "4": {
                        "hidden": true,
                        "notificationPreference": "hidden"
                    }
                },
                "permissions": [],
                "policyAvatar": null,
                "policyID": "5F78EE68DC8314F9",
                "policyName": null,
                "private_isArchived": "",
                "reportID": "2881610082725380",
                "reportName": "Expense Report #2881610082725380",
                "stateNum": 2,
                "statusNum": 4,
                "total": -2000,
                "tripData": null,
                "type": "expense",
                "unheldTotal": -2000,
                "visibility": null,
                "welcomeMessage": "",
                "writeCapability": "all"
            }
        },
        {
            "key": "report_616431942157016",
            "onyxMethod": "merge",
            "value": {
                "iouReportID": null
            }
        },
        {
            "key": "reportNextStep_2881610082725380",
            "onyxMethod": "set",
            "value": {
                "icon": "checkmark",
                "message": [
                    {
                        "text": "No further action required!"
                    }
                ],
                "nextReceiver": "",
                "requiresUserAction": false,
                "showForExpense": true,
                "showForMobile": true,
                "showUndoSubmit": false,
                "title": "Finished!",
                "type": "neutral"
            }
        }
    ],
    "jsonCode": 200,
    "requestID": "jgJeg1",
    "previousUpdateID": 10252,
    "lastUpdateID": 10268
}
```
</detials>

**Comment by roryabraham** [comment]
relevant front-end code: https://github.com/Expensify/App/blob/1db6abb66106b1a855321f907408357fbcc8f78c/src/components/LHNOptionsList/OptionRowLHN.tsx#L118

**Comment by roryabraham** [comment]
Ok, so I think that the `hasOutstandingChildRequest` Onyx update in `PayMoneyRequest` comes from [here](https://github.com/Expensify/Auth/blob/06c925d06fa6c15b4b496dcf3224de05e23199f9/auth/command/PayMoneyRequest.cpp#L585), which in turn comes from the giant `getChats` function. In there, I think I might need to unpack [ReportItem::isAwaitingActionFromCurrentUser](https://github.com/Expensify/Auth/blob/87b1834307c28e379f71c52fc4fa7c5ca5a56dd7/auth/lib/ReportItem.cpp#L29) a bit more

**Comment by roryabraham** [comment]
Update: I think the linked IOU is not being found [here](https://github.com/Expensify/Auth/blob/c30ea84e5f7f510124185bc2ddbda8b4329c5d60/auth/lib/Report.cpp#L8196) when it should be. Testing in focus mode right now, but also will need to test in most recent mode too.

Also, maybe this can be DRYed up but this is performance-sensitive code so I might default to trying to just get this working rather than rocking the boat too much.

**Comment by roryabraham** [comment]
@thienlnam volunteered to grab this one from me since tomorrow's my last day before parental leave and I'm focusing on trying to complete (or almost complete) CP to Prod

**Comment by trjExpensify** [comment]
Nice one, thank you Jack!

**Comment by thienlnam** [comment]
Been sidetracked wrapping up things for Hubspot doc, but will look into this in the next couple o days

**Comment by thienlnam** [comment]
Summarizing some things in the thread for my own investigation

Auth flow
1. PayMoneyRequest is called [here](https://github.com/Expensify/Auth/blob/06c925d06fa6c15b4b496dcf3224de05e23199f9/auth/command/PayMoneyRequest.cpp#L585), which in turn calls getChats and then proceeds to structure the report for onyx.
2. The way `hasOutstandingChildRequest` is calculated is via [ReportItem::isAwaitingActionFromCurrentUser](https://github.com/Expensify/Auth/blob/b9dca7a6f8b038ef2c167ec736658374b91179b5/auth/lib/ReportItem.cpp#L29C18-L29C49)
3. So I need to investigate whether the linkedReport is showing up as part of getLinkedReports ([here](https://github.com/Expensify/Auth/blob/c30ea84e5f7f510124185bc2ddbda8b4329c5d60/auth/lib/Report.cpp#L8196)), or why isAwaitingActionFromCurrentUser is not triggering

**Comment by thienlnam** [comment]
Unassigning since I've got other things on my plate I need to clear out

**Comment by cristipaval** [comment]
I'll investigate this tomorrow

**Comment by cristipaval** [comment]
Interesting that currently, I get no GBR at all even before paying the expenses.

**Comment by cristipaval** [comment]
I have a [draft PR](https://github.com/Expensify/Auth/pull/13193) that fixes the issue. It contains changes in `SubmitReport` command used by both NewDot and OldDot. This means that I'll ask more people for rigorous reviews to avoid any unexpected impact. So, what's left to get the PR merged is:

- Ask for reviews from people who historically contributed to the changed logic.
- Write automated tests for GBR on policy chats in workspaces with delayed submissions

**Comment by cristipaval** [comment]
The backend is fixed. @allroundexperts could you please verify whether there is any other work needed on the front-end?

**Comment by allroundexperts** [comment]
@cristipaval Frontend changes are needed as per the selected proposal. We just need to assign this to @tienifr as per https://github.com/Expensify/App/issues/34767#issuecomment-1931537728

**Comment by cristipaval** [comment]
thanks @allroundexperts

**Comment by cristipaval** [comment]
@tienifr, I just reproduced the issue in staging. I'll investigate it again.

**Comment by cristipaval** [comment]
I'm investigating this one today. Will post an update again in a few hours.

**Comment by cristipaval** [comment]
I started a discussion [here](https://expensify.slack.com/archives/C06ML6X0W9L/p1733824105239289) because I'm not sure if we want the GBR in this situation.

**Comment by cristipaval** [comment]
I'm working on a draft PR^

**Comment by cristipaval** [comment]
Work in progress.

**Comment by cristipaval** [comment]
Will continue on this tomorrow

**Comment by cristipaval** [comment]
this is my top priority today

**Comment by cristipaval** [comment]
I fixed the GBR but I still need to make the chat be returned in the OpenApp command as well, as a report that requires attention.

**Comment by cristipaval** [comment]
Update:
I have [this Auth PR](https://github.com/Expensify/Auth/pull/13737) that fixes the issue and also fixes the `Report::getChats` function to show the GBR in the LHN when the reimburser freshly signs in. The fix has been implemented, but I'll ask for some early reviews because any changes on the getChats might introduce performance issues. If I reach a consensus on the changes, I will write the automated tests and prepare the PR for merge.

**Comment by cristipaval** [comment]
Next step is to create time the new query in the getChats function for 5 busy accounts and time it.

**Comment by cristipaval** [comment]
I'll continue today for a couple hours on this one.

**Comment by cristipaval** [comment]
this is becoming my top priority again, will post an update later today or tomorrow

**Comment by muttmuure** [comment]
Going to demote this to MEDIUM to focus on higher impact perf issues

**Comment by cristipaval** [comment]
I'm working on the automated tests

**Comment by cristipaval** [comment]
[The Auth PR](https://github.com/Expensify/Auth/pull/13737) is ready for review

**Comment by cristipaval** [comment]
The PRs are in review.

**Comment by cristipaval** [comment]
[Auth PR](https://github.com/Expensify/Auth/pull/13737) is still in review

**Comment by cristipaval** [comment]
Daily Update:

- [Auth PR](https://github.com/Expensify/Auth/pull/13737) is merged
- [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/45044) is ready for merge, but held on the Auth PR

**Comment by cristipaval** [comment]
Waiting for the Auth PR to be deployed.

**Comment by cristipaval** [comment]
Web-E PR is in staging. On you now, @tienifr

**Comment by cristipaval** [comment]
<img width="765" alt="Image" src="https://github.com/user-attachments/assets/633fa109-078a-48b8-8391-27e53b8253fe" />

@tienifr, I see you approved the PR, which means that you're not following the reproduction steps. It should be Submit And Close. So the Workflows page should look like this:

<img width="733" alt="Image" src="https://github.com/user-attachments/assets/b9df4c6a-3bfc-4ec8-813f-32e6623838eb" />

BUT, I do agree that GBR should be shown in your case as well when the approvals are enabled. I'll investigate this new use case.

**Comment by cristipaval** [comment]
I couldn't reproduce the issue locally, even with the Approvals enabled 🤔

**Comment by cristipaval** [comment]
ok, then it works as expected. If you don't have ACH reimbursements enabled, then you have indirect reimbursements and we don't show GBR in that case.

**Comment by cristipaval** [comment]
@tienifr, `hasOutstandingChildRequest` has to be true for the expense chat, not the expense report itself.

So what's left as work on the front-end here is to:
- Verify that `PayMoneyRequest` call returns: 
    - `hasOutstandingChildRequest` = true for the parent chat when there are other reports awaiting reimbursements in the chat (this is to confirm that the backend is fixed)
    - `hasOutstandingChildRequest` = false otherwise
- Fix the optimistic Onyx data to not set `hasOutstandingChildRequest` according to the above logic

**Comment by cristipaval** [comment]
> I believe the next step on this issue is to implement the second solution, is it correct?

Correct, just fix the optimistic Onyx data, @tienifr

**Comment by cristipaval** [comment]
The fix PR is merged. It will be included in the next deploy

**Comment by allroundexperts** [comment]
@trjExpensify This issue was worth $500. I think melvin automatically changed it to $250 once the issue got moved from internal to external. Can you check?

**Comment by trjExpensify** [comment]
I can see that it was created as $500 in Jan 2024. Remind me what the rule change was with pricing? PRs in review, or issue created? 

P.S it's $0 until that sweet, sweet checklist gets filled out man. 😛

**Comment by trjExpensify** [comment]
>This issue was created before the date so should stay at $500

Sounds good!

**Comment by slafortune** [comment]
<img width="978" alt="Image" src="https://github.com/user-attachments/assets/549ecbe0-3d6c-4fe5-b7b9-3dcd07fe66c8" />

<img width="887" alt="Image" src="https://github.com/user-attachments/assets/1dd22ee7-3aa8-4117-b87f-3cd51d60ee3a" />

@allroundexperts Role C+  requires payment through NewDot Manual Requests - owed $500
@tienifr Role Contributor -  requires payment through UpWorks - offer sent here for $500 https://www.upwork.com/nx/wm/offer/106622769

**Comment by allroundexperts** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [x] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: This required a backend fix, and wasn't a regression caused by any frontend PR.

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: N/A

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.



- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


- Go to the workspace settings > More features > Enable Workflows
- Go to the Workflows tab in the LHN
- Enable Delay submissions
- Keep the default workflow of Submit and Close
- Enable ACH reimbursements set up for the workspace.

### Test:


1. [Employee] Create a manual request in the workspace chat and submit it
2. [Employee] Create another manual request and submit it
3. [Admin] Go to workspace chat with employee
4. [Admin] Pay the second submitted report (submitted in Step 2)
5. [Admin] Click on the preview of first submitted report (Step 1) and return to the main chat

Verify that: In Step 4, green dot should still appear for the workspace chat in LHN because there is still unpaid report

Do we agree 👍 or 👎

**Comment by slafortune** [comment]
@allroundexperts Role C+  requires payment through NewDot Manual Requests - Owed $500
@tienifr Role Contributor - requires payment through Upworks - Paid $500 ✅

**Comment by JmillsExpensify** [comment]
$500 approved for @allroundexperts
